File: Examples/VB/HelloWorld/HelloWorld.vb
Imports StateForge.StateMachine


Namespace Company.Product

	Partial Public Class HelloWorld
		Private context As HelloWorldContext

		Public Sub New()
			context = New HelloWorldContext(Me)
			Me.context.Observer = ObserverConsole.Instance
		End Sub

		Protected Friend Sub DoPrint()
			Console.WriteLine("HelloWorld")
		End Sub

		Shared Sub Main(ByVal args() As String)
			Dim helloWorld_Renamed = New HelloWorld()
			helloWorld_Renamed.EvPrint()
			Environment.Exit(0)
		End Sub
	End Class
End Namespace

File: Examples/CS/HelloWorld/HelloWorld.cs

namespace StateForge.Examples
{
    using System;
    using StateForge.StateMachine;

    public partial class HelloWorld
    {
        private HelloWorldContext context;

        public HelloWorld()
        {
            this.context = new HelloWorldContext(this);
            this.context.Observer = ObserverConsole.Instance;
        }

        protected internal void DoPrint()
        {
            Console.WriteLine("HelloWorld");
        }

        static void Main(string[] args)
        {
            var helloWorld = new HelloWorld();
            helloWorld.EvPrint();
            Environment.Exit(0);
        }
    }
}
